modules/components/src/Arranger/MatchBox.js

```diff
@@ -245,28 +245,23 @@ const MatchBox = ({
                       }),
                       dataPath,
                     );
-                    if (setSQON) {
-                      setSQON(
-                        toggleSQON(
+                    const nextSQON = toggleSQON(
+                      {
+                        op: 'and',
+                        content: [
                           {
-                            op: 'and',
-                            content: [
-                              {
-                                op: 'in',
-                                content: {
-                                  field: primaryKeyField?.field,
-                                  value: [].concat(
-                                    `set_id:${data.setId}` || [],
-                                  ),
-                                },
-                              },
-                            ],
+                            op: 'in',
+                            content: {
+                              field: primaryKeyField?.field,
+                              value: [].concat(`set_id:${data.setId}` || []),
```

Comment:
would ``set_id:${data.setId}` || []` ever evaluate to `[]`? if `data.setId` is the null check then maybe `data.setId ? `set_id:${data.setId}` : []` is a safer check

Reply:
`[...(data.setId ? ['set_id:${data.setId}'] : [])]` I guess